### PR TITLE
RN 0.57

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] circle
- [x] prod desktop works
- [x] rn-storybook works (so the web server doesn't work but the on device UI works so we have to use taht for now. )
- [x] maybe remove some polyfills from globals.native
- [x] jsc android
- [x] desktop works
- [x] ios works
- [x] terminal packager
- [x] android works
- [x] prod android works
- [x] prod ios works
- [x] storybook works
- [x] jest works

### CODE REVIEW

File: shared/app/globals.native.js [hunk 1]
```diff
@@ -18,12 +17,4 @@ if (typeof __STORYBOOK__ === 'undefined') {
 // We don't storyshot RN
 __STORYSHOT__ = false
 
-// Native String.startswith() sometimes incorrectly returns false on Android!
```
Comment (chrisnojima) [inline shared/app/globals.native.js]: no longer needed with modern jsc

File: shared/settings/passphrase/index.js [hunk 2]
```diff
@@ -70,8 +70,9 @@ class UpdatePassphrase extends Component<Props, State> {
           hintText="New passphrase"
           type={inputType}
           errorText={this.props.newPassphraseError}
+          value={this.state.passphrase}
           onChangeText={passphrase => this._handlePassphraseChange(passphrase)}
-          uncontrolled={true}
+          uncontrolled={false}
```
Comment (chrisnojima) [inline shared/settings/passphrase/index.js]: just to prove this works

**Comment by akalin-keybase** [comment]
CircleCI passes, but it's not using the updated android stuff. still working on that...

**Comment by akalin-keybase** [comment]
CircleCI should be working with the updated Android stuff now -- see https://github.com/keybase/client/pull/13400/files

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump. they released 0.57 final

File: .circleci/config.yml [hunk 3]
```diff
@@ -24,7 +24,7 @@ jobs:
     # Install JS dependencies.
     - run:
         working_directory: shared
-        command: yarn install --pure-lockfile --ignore-engines --prefer-offline
```
Comment (chrisnojima) [inline .circleci/config.yml]: workaround to https://github.com/yarnpkg/yarn/issues/2629

File: shared/jenkins_test.sh [hunk 4]
```diff
@@ -48,7 +48,7 @@ js_tests() {
 
     echo 'yarn install'
     yarn cache clean
-    yarn install --pure-lockfile --prefer-offline --no-emoji --no-progress
+    yarn install --pure-lockfile --no-emoji --no-progress --network-concurrency 1
```
Comment (chrisnojima) [inline shared/jenkins_test.sh]: workaround to https://github.com/yarnpkg/yarn/issues/2629

File: shared/app/main.native.js [hunk 5]
```diff
@@ -46,14 +46,15 @@ class Main extends React.Component<Props> {
...
-          component={View}
+          component={ViewForGatewayDest}
           pointerEvents="box-none"
           style={globalStyles.fillAbsolute}
         />
       </React.Fragment>
     )
   }
 }
+const ViewForGatewayDest = (props: any) => <View {...props} />
```
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/app/main.native.js]: What does, flow?

[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/app/main.native.js]: no, react native, crashes w/o this

File: shared/chat/conversation/messages/attachment/image/image-render.native.js [hunk 6]
```diff
@@ -38,18 +37,9 @@ export class ImageRender extends React.Component<Props> {
...
-          ref={ref => {
-            this.webview = ref
-          }}
-          styles={this.props.style}
-          onLoadEnd={this._allLoads}
-          source={source}
-          scrollEnabled={false}
-        />
-      ) : (
+      return (
```
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/chat/conversation/messages/attachment/image/image-render.native.js]: Also it fails in a different way on the android emulator

File: shared/babel.config.js [hunk 7]
```diff
@@ -4,16 +4,35 @@
...
+  if (!isElectron && !isReactNative) {
+    throw new Error('MUST have env var BABEL_PLATFORM to all babel')
+  }
+  if (isElectron && isReactNative) {
+    throw new Error('Packager is confused about babel platform')
   }
 
-  return config
+  if (isElectron) {
+    console.error('KB babel.config.js for Electron')
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/babel.config.js]: so you can see it, in some contexts they hide logs and whatever so this was the best choice i think

File: shared/package.json [hunk 8]
```diff
@@ -226,28 +227,104 @@
...
     "fs-extra": "6.0.1",
-    "jest": "23.1.0",
-    "jest-glamor-react": "4.3.0",
+    "jest": "23.5.0",
+    "jsc-android": "^216113.0.0",
+    "jest-glamor-react": "4.3.1",
     "json5": "1.0.1",
     "klaw-sync": "4.0.0",
     "minimist": "1.2.0",
+    "msgpack": "git://github.com/keybase/nullModule",
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/package.json]: before we had a postinstall process to empty it out, instead we just point to this instead, a little bit cleaner

File: .circleci/config.yml (hunk 3 above)
Comment (akalin-keybase) [reply .circleci/config.yml]: Can you reference the issue in a comment?

Comment (chrisnojima) [reply .circleci/config.yml]: what do you mean?

Comment (akalin-keybase) [reply .circleci/config.yml]: can you add a comment linking to https://github.com/yarnpkg/yarn/issues/2629 as a comment in the file itself

**Comment by akalin-keybase** [comment]
tested some text input stuff on android, looks good!

**Pull request merged**